### PR TITLE
🎉Release 2.0.0

### PR DESCRIPTION
**Pull request by Lebyy**
This release fixes tons of bugs and issues with the package and has completely re-written and enhanced code base.
~~**NOTE:** I don't recommend you to upgrade yet since this is a work in progress pr and it will update frequently with many changes to the save system, storage etc.~~ 
Feel free to install this branch by `npm i Lebyy/discord-voice#rewrite`. 
~~Currently there are little to no docs present and are on the todo list.~~

# Planned Additions / Changes

## ✅ Done

- [x] Live Voice Tracking. 
- [x] Support for multiple databases. 
- [x] All previous bug's fixed.
- [x] Will be moving to JSON db as default.
- [x] Sharding support. 
- [x] Entire code-base re-write.
- [x] Voice Time Based XP/Levels.
- [x] Better and Easy docs.
- [x] Tutorials in the examples folder.

### CODE REVIEW

**Comment by Lebyy** [comment]
1. Live Voice Tracking. - [x] Done.
2. Support for multiple databases. - [x] Done.
3. Entire code-base re-write. - [] ⚠️ In Progress.
4. All previous bug's fixed. - [x] Done.
5. Better and Easy docs. - [] ⚠️ In Progress.
6. Tutorials in the TUTORIALS.md file. - [] ⚠️ In Progress.
7. Will be moving to JSON db as deafult. - [x] Done.
8. Sharding support. - [x] Done.
9. Voice Time Based XP/Levels. - [] ⚠️ In Progress.

**Review by DeltaCoderr — APPROVED** [review]

File: src/Manager.js [hunk 1]
```diff
@@ -0,0 +1,286 @@
...
+    if (!client) throw new Error("Client is a required option.");
+    this.client = client;
+    this.ready = false;
+    this.users = [];
+    this.configs = [];
+    this.options = merge(defaultManagerOptions, options);
+    if (init) this._init();
+  }
+
+  async updateConfig(guildID, updatedConfig) {
```
Comment (Koolwiza) [inline src/Manager.js:23]:
```suggestion
 updateConfig(guildID, updatedConfig) {
```
no use for async? It is redundant

Comment (Lebyy) [reply src/Manager.js:23]: Yep tysm!

File: src/Manager.js [hunk 2]
```diff
@@ -0,0 +1,286 @@
...
+
+  async saveConfig(guildID, configData) {
+    await writeFileAsync(
+      this.options.configStorage,
+      JSON.stringify(this.configs.map((config) => config.data)),
+      "utf-8"
+    );
+    this.refreshConfigStorage();
+    return;
+  }
```
Comment (Koolwiza) [inline src/Manager.js:101]: How does this work? I am a bit confused, you are not using any of the params...

Comment (Lebyy) [reply src/Manager.js:101]: So basically the params will be used for the custom db's the user's provide for the default json db we used the configs and users array.

Comment (Koolwiza) [reply src/Manager.js:101]: Uh ok, but why do you need them

Comment (Lebyy) [reply src/Manager.js:101]: To save the data.

**Comment by Lebyy** [comment]
We are almost ready for release! You can leave your reviews about this pr and report the bug's which occur during your testing.

**Review by Instinzts — APPROVED** [review]
Based on the email i received about the bug's we encounter, some errors where based on Discord.jsV13 which changed guildID to guildId.

**Comment by Lebyy** [comment]
> Based on the email i received about the bug's we encounter, some errors where based on Discord.jsV13 which changed guildID to guildId.

Those were fixed in the last group of commits :D

**Comment by Lebyy** [comment]
discord-voice@2.0.0 has been released to npm with the dev tag!
You can find it [here](https://www.npmjs.com/package/discord-voice/v/2.0.0).

To install it you can run `npm i --save discord-voice@dev`.

Docs are located at: https://discord-voice.js.org/
(The links in the readme point to the master branch for now you can replace master to rewrite in the url's to get access to the link).

**Review by DeltaCoderr — APPROVED** [review]

**Pull request merged**